graph/path/yen_ksp_test.go

```diff
@@ -0,0 +1,64 @@
+// Copyright ©2018 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package path
+
+import (
+	"math"
+	"testing"
+
+	"gonum.org/v1/gonum/graph/simple"
+	"gonum.org/v1/gonum/graph"
+)
+
+
+func TestYenKSP(t *testing.T) {
```

Comment:
I'd like to see more test cases here, so a table of test cases would be the way to go. At the minimum, we need some cases where there are other implementations or obviously correct solutions. I would also like to see some test cases where there are fewer than k possible paths from s to t to see how the implementation deals with that, and cases where there are k equally weighted paths and k paths with some more costly paths before the k runout.

Reply:
Yeah, I am working on moving the test cases to a table, and then I will add some more. Thanks for all the help with this.